addon/services/session.js

```diff
@@ -212,16 +212,31 @@ export default Ember.Service.extend(Ember.Evented, {
     });
     ```
 
+    This function accepts any number of arguments but two at minimum. The first argument is
+    always the name of the used authorizer and the last argument is always a function (`block`) to
+    be called with the authorization data generated by the authorizer. Between those two arguments,
+    zero or more additional arguments can be given and they will be bassed on as is to the authorizer
+    when called.
+
     @method authorize
     @param {String} authorizer The authorizer to authorize the block with
     @param {Function} block The block of code to call with the authorization data generated by the authorizer
     @public
   */
-  authorize(authorizerFactory, block) {
+  authorize(...args) {
+    Ember.assert('session.authorize() must be called with at least two parameters', args.length >= 2);
+    Ember.assert('First parameter to session.authorize() must be string with the name of the used authorizer', typeof args[0] === 'string');
+    Ember.assert('Second parameter to session.authorize() must be function', typeof args[args.length - 1] === 'function');
```

Comment:
I added two assertions here. What do you think?

I figured that since we now support any number of arguments but still make assumptions about the first argument and the last argument, it would probably make it easier for developers to fail early with decent error message.